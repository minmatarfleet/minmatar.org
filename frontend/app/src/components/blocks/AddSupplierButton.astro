---
import { i18n } from '@helpers/i18n'
const { t, translatePath } = i18n(Astro.url)

const ADD_SUPPLIER_TOKEN_INIT_PARTIAL_URL = `${translatePath('/redirects/add_supplier_token_init')}?redirect=${Astro.url}`

import Button from '@components/blocks/Button.astro'
---

<Button
    size='sm'
    color='green'
    href={ADD_SUPPLIER_TOKEN_INIT_PARTIAL_URL}
    x-data={`{
        redirect(accepted) {
            if (accepted) window.location.href = "${ADD_SUPPLIER_TOKEN_INIT_PARTIAL_URL}"
        },
        show_become_supplier_disclaimer() {
            show_confirm_dialog({
                title: '${t('become_supplier_dialog_title')}',
                partial: '${translatePath('/partials/become_supplier_dialog/')}',
            }).then( (accepted) => this.redirect(accepted) )
        }
    }`}
    x-on:click.prevent="show_become_supplier_disclaimer"
>
    {t('add_supplier')}
</Button>