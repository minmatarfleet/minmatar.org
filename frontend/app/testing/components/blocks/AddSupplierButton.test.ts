import { experimental_AstroContainer as AstroContainer } from "astro/container";
import { expect, test } from "vitest";
import AddSupplierButton from "@components/blocks/AddSupplierButton.astro";

test("AddSupplierButton defaults", async () => {
  const container = await AstroContainer.create();
  const result = await container.renderToString(AddSupplierButton, {});

  expect(result).toMatchSnapshot();
});
