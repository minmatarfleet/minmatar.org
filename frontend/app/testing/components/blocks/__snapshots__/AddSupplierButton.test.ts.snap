// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddSupplierButton defaults 1`] = `
"<a href="/redirects/add_supplier_token_init?redirect=https://example.com/" x-data="{
        redirect(accepted) {
            if (accepted) window.location.href = &#34;/redirects/add_supplier_token_init?redirect=https://example.com/&#34;
        },
        show_become_supplier_disclaimer() {
            show_confirm_dialog({
                title: 'Become a supplier',
                partial: '/partials/become_supplier_dialog/',
            }).then( (accepted) => this.redirect(accepted) )
        }
    }" x-on:click.prevent="show_become_supplier_disclaimer" data-size="sm" style="--button_background_color: var(--green); --button_color: var(--fleet-yellow)" data-astro-cid-r5e2wo56="true" class="[ button ]"> <div class="flex-inline" data-astro-cid-r5e2wo56="true" data-astro-cid-ud5fqf5k style="--flex_inline_gap: var(--space-xs);--flex_inline_justification: center;">   Add supplier entity  </div>   </a> "
`;
